held_suarez/held_suarez.F90

```diff
@@ -0,0 +1,166 @@
+    psurf_ref     = psurf_ref_in
+    pref_mid_norm = pref_mid_norm_in
+
+  end subroutine held_suarez_1994_init
+
+  !> \section arg_table_held_suarez_1994_run  Argument Table
+  !! \htmlinclude held_suarez_1994_run.html
+  subroutine held_suarez_1994_run(ncol, clat, pmid, &
+       u, v, t, du, dv, s)
+
+    !
+    ! Input arguments
+    !
+    integer,  intent(in)  :: ncol             ! Num active columns
+    real(kind_phys), intent(in)  :: clat(:)      ! latitudes(radians) for columns
+    real(kind_phys), intent(in)  :: pmid(:,:) ! mid-point pressure
+    real(kind_phys), intent(in)  :: u(:,:)    ! Zonal wind (m/s)
+    real(kind_phys), intent(in)  :: v(:,:)    ! Meridional wind (m/s)
+    real(kind_phys), intent(in)  :: t(:,:)    ! Temperature (K)
+                                              !
+                                              ! Output arguments
+                                              !
+    real(kind_phys), intent(out) :: du(:,:)   ! Zonal wind tend
+    real(kind_phys), intent(out) :: dv(:,:)   ! Meridional wind tend
+    real(kind_phys), intent(out) :: s(:,:)    ! Heating rate
+    !
+    !---------------------------Local workspace-----------------------------
+    !
+    integer  :: i, k          ! Longitude, level indices
+
+    real(kind_phys) :: kv            ! 1./efolding_time (normalized) for wind
+    real(kind_phys) :: kt            ! 1./efolding_time for temperature diss.
+    real(kind_phys) :: trefa         ! "radiative equilibrium" T
+    real(kind_phys) :: trefc         ! used in calc of "radiative equilibrium" T
+    real(kind_phys) :: cossq(ncol)   ! coslat**2
+    real(kind_phys) :: cossqsq(ncol) ! coslat**4
+    real(kind_phys) :: sinsq(ncol)   ! sinlat**2
+    real(kind_phys) :: coslat(ncol)  ! cosine(latitude)
+    !
+    !-----------------------------------------------------------------------
+    !
+
+    do i = 1, ncol
+      coslat (i) = cos(clat(i))
+      sinsq  (i) = sin(clat(i))*sin(clat(i))
+      cossq  (i) = coslat(i)*coslat(i)
+      cossqsq(i) = cossq (i)*cossq (i)
+    end do
+
+    !
+    !-----------------------------------------------------------------------
+    !
+    ! Held/Suarez IDEALIZED physics algorithm:
+    !
+    !   Held, I. M., and M. J. Suarez, 1994: A proposal for the
+    !   intercomparison of the dynamical cores of atmospheric general
+    !   circulation models.
+    !   Bulletin of the Amer. Meteor. Soc., vol. 75, pp. 1825-1830.
+    !
+    !-----------------------------------------------------------------------
+    !
+    ! Compute idealized radiative heating rates (as dry static energy)
+    !
+    !
+    do k = 1, pver
+      if (pref_mid_norm(k) > sigmab) then
+        do i = 1, ncol
+          kt = ka + (ks - ka)*cossqsq(i)*(pref_mid_norm(k) - sigmab)/onemsig
+          trefc   = 315._kind_phys - (60._kind_phys * sinsq(i))
+          trefa = (trefc - 10._kind_phys*cossq(i)*log((pmid(i,k)/psurf_ref)))*(pmid(i,k)/psurf_ref)**cappa
```

Comment:
I would convert the `10._kind_phys` number into its own parameter, as it is representing a specific, potentially tunable, physical quantity (the change in potential temperature over one scale height of the atmosphere), at least according to the original manuscript.